talisker/metrics.py

```diff
@@ -183,8 +192,15 @@ def prometheus_cleanup_worker(pid):
 
     histogram_path = os.path.join(prom_dir, histogram_archive)
     counter_path = os.path.join(prom_dir, counter_archive)
+    archive_paths = [histogram_path, counter_path]
+    archive_paths = [p for p in archive_paths if os.path.exists(p)]
 
-    metrics = collect(paths + [histogram_path, counter_path])
+    collect_paths = paths + archive_paths
+    try:
+        collector = MultiProcessCollector(None)
```

Comment:
pull this out of the `try..except..` since we don't expect it to raise `AttributeError`

Reply:
ok